NumericUI/scripts/mods/NumericUI/Interactions.lua

```diff
@@ -0,0 +1,114 @@
+local INTERACTIONS_HUD_DEF_PATH = "scripts/ui/hud/elements/interaction/hud_element_interaction_definitions"
+
+local backups = mod:persistent_table("interactions_hud_backups")
+backups.definitions = backups.definitions or table.clone(mod:original_require(INTERACTIONS_HUD_DEF_PATH))
+
+local UIWidget = require("scripts/managers/ui/ui_widget")
+local UIHudSettings = require("scripts/settings/ui/ui_hud_settings")
+local Pickups = require("scripts/settings/pickup/pickups")
+
+local current_ammo = nil
+local max_ammo = nil
+local small_clip_gain = 0
+local large_clip_gain = 0
+
+mod:hook_require(INTERACTIONS_HUD_DEF_PATH, function(instance)
+	
+	local ammo_description_size_mod = 0.9
+	local description_text_style = table.clone(backups.definitions.widget_definitions.description_text.style.text)
+
+	instance.widget_definitions.ammo_loss_description_text = UIWidget.create_definition({
+		{
+			value_id = "text",
+			style_id = "text",
+			pass_type = "text",
+			value = " ",
+			retained_mode = false,
+			style = table.merge_recursive(table.clone(description_text_style), {
+				font_size = description_text_style.font_size * ammo_description_size_mod,
+				text_color = UIHudSettings.color_tint_ammo_high,
+				text_horizontal_alignment = "center",
+				text_vertical_alignment = "bottom",
+				offset = {0,-6,0}
+				
+			}),
+		},
+	}, "description_box")
+end)
+
+
+mod:hook_safe("HudElementInteraction", "_setup_interaction_information", function(self, interactee_unit, interactee_extension, interactor_extension)
+	
+	if mod:get("show_ammo_amount_from_packs") then
+		local interaction_widget = self._widgets_by_name.description_text
+
+		if interaction_widget and max_ammo then
+			local ammo_gain_widget = self._widgets_by_name.description_text
+			local ammo_loss_widget = self._widgets_by_name.ammo_loss_description_text
+			local hud_description = interactor_extension:hud_description()
+
+			local missing_ammo = max_ammo - current_ammo
+			local ammo_gain_text = ""
+			local ammo_loss_text = ""
+
+			if hud_description == "loc_pickup_consumable_small_clip_01" then
+				if missing_ammo >= small_clip_gain then
+					ammo_gain_text = string.format("  +%d", small_clip_gain)
+					
+
+				elseif missing_ammo > 0 then
+					ammo_gain_text = string.format("  +%d", missing_ammo)
+					ammo_loss_text = string.format("(%d)", (small_clip_gain - missing_ammo))
+					ammo_loss_widget.style.text.offset[1] = 15*string.len(missing_ammo)
+				end
+
+			elseif hud_description == "loc_pickup_consumable_large_clip_01" then
+				if missing_ammo >= large_clip_gain then
+					ammo_gain_text = string.format("  +%d", large_clip_gain)
+
+				elseif missing_ammo > 0 then
+					ammo_gain_text = string.format("  +%d", missing_ammo)
+					ammo_loss_text = string.format("(%d)", (large_clip_gain - missing_ammo))
+					ammo_loss_widget.style.text.offset[1] = 15*string.len(missing_ammo) + 30 --The extra 30 is the size difference between "Ammo Tin" and "Ammo Pack"
```

Comment:
These string.len calculations could potentially use UIRenderer.text_size, that would negate the need for the extra + 30 and it would also better support localization (e.g. the chinese text is probably a different width).

Not sure how it'd fit into this function but maybe we can just calculate the widths up front